spec/services/register_and_partner_api/sync_users_spec.rb

```diff
@@ -23,5 +23,13 @@
       record = User.find_by(email: "school-leader@example.com")
       expect(record.full_name).to eql("Induction Tutor")
     end
+
+    it "updates an existing user that has changed their email address" do
+      described_class.perform
+
+      record = User.find_by(register_and_partner_id: "65abf54c-c7c2-490b-9bcd-bbb4e0aab934")
+      expect(User.count).to eql(2)
+      expect(record.email).to eql("lead-provider-1@example.com")
+    end
```

Comment:
An easier way would be to set up a user in the test, before you call `perform`, and give it a different email but the same id. 

Then you can avoid the duplicated user in our `users.json` file :)